addons/rook-ceph/enable

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+import click
+import os
+import subprocess
+import shutil
+from pathlib import Path
+
+# Script dir
+DIR = Path(__file__).absolute().parent
+
+# Snap paths
+SNAP = Path(os.getenv("SNAP") or "/snap/microk8s/current")
+SNAP_COMMON = Path(os.getenv("SNAP_COMMON") or "/var/snap/microk8s/common")
+KUBECTL = SNAP / "microk8s-kubectl.wrapper"
+HELM = SNAP / "microk8s-helm.wrapper"
+KUBELET_DIR = SNAP_COMMON / "var" / "lib" / "kubelet"
+
+# Addon specifics
+ROOK_PLUGINS = [
+    "connect-external-ceph",
+    ".rook-create-external-cluster-resources.py",
+    ".rook-import-external-cluster.sh",
+]
+
+
+@click.command()
+@click.option("--rook-version", default="v1.11.9")
+@click.option("--rook-repository", default="https://charts.rook.io/release")
+@click.option("--set", "helm_set", multiple=True)
+@click.option("-f", "--values", "helm_values", multiple=True, type=click.Path(exists=True))
+def main(rook_version: str, rook_repository: str, helm_set: list, helm_values: list):
+    click.echo(f"Add Rook Helm repository {rook_repository}")
+    subprocess.run([HELM, "repo", "add", "rook-release", rook_repository])
+    subprocess.run([HELM, "repo", "update", "rook-release"])
+
+    helm_args = [
+        "install",
+        "rook-ceph",
+        "rook-release/rook-ceph",
+        f"--version={rook_version}",
+        "--create-namespace",
+        "--namespace=rook-ceph",
+        f"--set=csi.kubeletDirPath={KUBELET_DIR}",
+    ]
+
+    for arg in helm_set:
+        helm_args.extend(["--set", arg])
+    for file in helm_values:
+        helm_args.extend(["--values", file])
+
+    click.echo(f"Install Rook version {rook_version}")
+    subprocess.run([HELM, *helm_args])
+
+    click.echo("Creating 'microk8s connect-external-ceph' command")
+    for plugin in ROOK_PLUGINS:
+        try:
+            shutil.copyfile(DIR / "plugin" / plugin, SNAP_COMMON / "plugins" / plugin)
```

Comment:
I go this error:
```
ubuntu@ip-172-31-3-156:~$ sudo microk8s connect-external-ceph
/snap/microk8s/x1/microk8s.wrapper: line 40: /var/snap/microk8s/common/plugins/connect-external-ceph: Permission denied
```
I think the plugin needs to be set to executable after

Reply:
Interestingly, copyfile does not preserve mode bits, and it makes sense looking at the docs. Fixed
```py
In [13]: shutil.copy('./connect-external-ceph', 'a')
Out[13]: 'a'

In [14]: shutil.copyfile('./connect-external-ceph', 'b')
Out[14]: 'b'

In [15]: ls -la a b connect-external-ceph
-rwxrwxr-x 1 ubuntu ubuntu 5610 Ιουλ 19 14:00 a*
-rw-rw-r-- 1 ubuntu ubuntu 5610 Ιουλ 19 14:00 b
-rwxrwxr-x 1 ubuntu ubuntu 5610 Ιουλ 19 09:12 connect-external-ceph*
```